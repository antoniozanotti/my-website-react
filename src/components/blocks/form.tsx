import React, { useRef, useState } from "react";
import { TzButton, TzInput, TzTextarea } from "topaz-react";
import emailjs from "@emailjs/browser";
import ReCAPTCHA from "react-google-recaptcha";

export default function Form() {
  const form = useRef<HTMLFormElement>(null);
  const fromName = useRef<HTMLInputElement>(null);
  const [fromNameValue, setFromNameValue] = useState("");
  const [passedRecaptcha, setPassedRecaptcha] = useState(false);
  const [status, setStatus] = useState("typing");
  const recaptchaRef = useRef<ReCAPTCHA>(null);

  const onChangeRecaptcha = () => {
    setPassedRecaptcha(true);
  };

  const handleSubmit = (e: React.SyntheticEvent) => {
    e.preventDefault();
    if(!passedRecaptcha){
      return false;
    }
    setStatus("submitting");
    setFromNameValue(fromName.current?.value ?? "");
    emailjs
      .sendForm("service_7vfby3r", "template_a9dy2ok", form.current ?? "", {
        publicKey: "VLh3o8U_wS41PEp8F",
      })
      .then(
        () => {
          setStatus("success");
          form.current?.reset();
        },
        (error) => {
          setStatus("error");
        }
      );
  };

  return (
    <form
      id="form_lets_work_together"
      className="flex flex-col space-y-4"
      ref={form}
      onSubmit={handleSubmit}
    >
      <p className="mt-4">
        I would like to be part of a team whose people respect each other,
        listen to each other and who work together to achieve clear goals. Leave a message, thank you in advance for your
        attention.
      </p>
      <label className="text-primary-2 dark:text-dark-primary-2 flex flex-col space-y-1">
        <span className="px-[10px] sm:px-[14px] lg:px-[20px]">Name</span>
        <TzInput
          type="text"
          variant="secondary"
          name="fromName"
          required
          ref={fromName}
        />
      </label>
      <label className="text-primary-2 dark:text-dark-primary-2 flex flex-col space-y-1">
        <span className="px-[10px] sm:px-[14px] lg:px-[20px]">Email</span>
        <TzInput type="email" variant="secondary" name="fromEmail" required />
      </label>
      <label className="text-primary-2 dark:text-dark-primary-2 flex flex-col space-y-1">
        <span className="px-[10px] sm:px-[14px] lg:px-[20px]">Message</span>
        <TzTextarea rows={7} variant="secondary" name="message" required />
      </label>
      <ReCAPTCHA
        ref={recaptchaRef}
        sitekey="6LdXUoIpAAAAABMIeNv1tu2fa_ssuP8qYGKkLGBU"
        onChange={onChangeRecaptcha}
      />
      {status === "success" && (
        <p>
          Hi {fromNameValue}, I will respond within 24 hours.
          <br />
          Thank you for the message sent.
        </p>
      )}
      {status === "error" && (
        <p>Hi {fromNameValue}, something went wrong. Please try later..</p>
      )}
      <div className="flex place-content-end">
        <TzButton
          iconName="ArrowRightIcon"
          isIconAfterLabel
          variant="accent"
          label="Submit"
        />
      </div>
    </form>
  );
}
