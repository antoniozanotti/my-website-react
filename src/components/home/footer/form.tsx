"use client";
import React, { useRef, useState } from "react";
import { TzButton, TzInput, TzTextarea } from "topaz-react";
import emailjs from "@emailjs/browser";
import ReCAPTCHA from "react-google-recaptcha";

export default function Form() {
  const form = useRef<HTMLFormElement>(null);
  const fromName = useRef<HTMLInputElement>(null);
  const [fromNameValue, setFromNameValue] = useState("");
  const [status, setStatus] = useState("typing");
  const recaptchaRef = useRef<ReCAPTCHA>(null);
  const [recaptchaValue, setRecaptchaValue] = useState("");

  const onChangeRecaptcha = (token: string | null) => {
    //const token = recaptchaRef.current?.getValue() ?? "";
    setRecaptchaValue(token ?? "");
  };

  const handleSubmit = (e: React.SyntheticEvent) => {
    e.preventDefault();
    recaptchaRef.current?.execute();
    recaptchaRef.current?.reset();
    if (recaptchaValue == "") {
      return false;
    }
    setStatus("submitting");
    setFromNameValue(fromName.current?.value ?? "");
    emailjs
      .sendForm("service_7vfby3r", "template_a9dy2ok", form.current ?? "", {
        publicKey: process.env.NEXT_PUBLIC_EMAILJS_PUBLIC_KEY,
      })
      .then(
        () => {
          setStatus("success");
          form.current?.reset();
        },
        (error) => {
          setStatus("error");
        }
      );
  };

  const recaptchaKey = process.env.NEXT_PUBLIC_GOOGLE_RECAPTCHA ?? "";

  return (
    <form
      id="form_lets_work_together"
      className="flex flex-col space-y-4"
      ref={form}
      onSubmit={handleSubmit}
    >
      <label className="text-primary-2 dark:text-dark-primary-2 flex flex-col space-y-1">
        <span className="px-[10px] sm:px-[14px] lg:px-[20px]">Name</span>
        <TzInput
          type="text"
          variant="secondary"
          name="fromName"
          required
          ref={fromName}
        />
      </label>
      <label className="text-primary-2 dark:text-dark-primary-2 flex flex-col space-y-1">
        <span className="px-[10px] sm:px-[14px] lg:px-[20px]">Email</span>
        <TzInput type="email" variant="secondary" name="fromEmail" required />
      </label>
      <label className="text-primary-2 dark:text-dark-primary-2 flex flex-col space-y-1">
        <span className="px-[10px] sm:px-[14px] lg:px-[20px]">Message</span>
        <TzTextarea rows={7} variant="secondary" name="message" required />
      </label>
      <ReCAPTCHA
        ref={recaptchaRef}
        sitekey={recaptchaKey}
        size="invisible"
        onChange={onChangeRecaptcha}
      />
      {status === "success" && (
        <p>
          Hi {fromNameValue}, I will respond within 24 hours.
          <br />
          Thank you for the message sent.
        </p>
      )}
      {status === "error" && (
        <p>Hi {fromNameValue}, something went wrong. Please try later..</p>
      )}
      <div className="flex place-content-end">
        <TzButton
          iconName="ArrowRightIcon"
          isIconAfterLabel
          variant="accent"
          label="Submit"
        />
      </div>
      <input type="hidden" name="g-recaptcha-response" value={recaptchaValue} />
    </form>
  );
}
