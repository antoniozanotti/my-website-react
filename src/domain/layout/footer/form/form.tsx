"use client";
import { useRef, useState, SyntheticEvent } from "react";
import { TzButton, TzInput, TzTextarea } from "topaz-react";
import emailjs from "@emailjs/browser";
import ReCAPTCHA from "react-google-recaptcha";
import {useTranslations} from 'next-intl';

export function Form() {
  const t = useTranslations('Footer.form');
  const formRef = useRef<HTMLFormElement>(null);
  const fromNameRef = useRef<HTMLInputElement>(null);
  const [fromNameValue, setFromNameValue] = useState("");
  const [status, setStatus] = useState("typing");
  const recaptchaRef = useRef<ReCAPTCHA>(null);

  async function handleSubmit (e: SyntheticEvent) {
    e.preventDefault();
    const token = await recaptchaRef.current?.executeAsync() ?? "";
    if (token == "") {
      return false;
    }
    setStatus("submitting");
    setFromNameValue(fromNameRef.current?.value ?? "");
    emailjs
      .sendForm("service_7vfby3r", "template_a9dy2ok", formRef.current ?? "", {
        publicKey: process.env.NEXT_PUBLIC_EMAILJS_PUBLIC_KEY,
      })
      .then(
        () => {
          setStatus("success");
          formRef.current?.reset();
          recaptchaRef.current?.reset()
        },
        (error) => {
          setStatus("error");
        }
      );
  };

  const recaptchaKey = process.env.NEXT_PUBLIC_GOOGLE_RECAPTCHA ?? "";

  return (
    <form
      id="form_lets_work_together"
      className="flex flex-col space-y-4"
      ref={formRef}
      onSubmit={handleSubmit}
    >
      <label className="text-primary-2 dark:text-dark-primary-2 flex flex-col space-y-1">
        <span className="px-[10px] sm:px-[14px] lg:px-[20px]">{t('name')}</span>
        <TzInput
          type="text"
          variant="secondary"
          name="fromName"
          required
          ref={fromNameRef}
        />
      </label>
      <label className="text-primary-2 dark:text-dark-primary-2 flex flex-col space-y-1">
        <span className="px-[10px] sm:px-[14px] lg:px-[20px]">Email</span>
        <TzInput type="email" variant="secondary" name="fromEmail" required />
      </label>
      <label className="text-primary-2 dark:text-dark-primary-2 flex flex-col space-y-1">
        <span className="px-[10px] sm:px-[14px] lg:px-[20px]">{t('message')}</span>
        <TzTextarea rows={7} variant="secondary" name="message" required />
      </label>
      <ReCAPTCHA
        ref={recaptchaRef}
        sitekey={recaptchaKey}
        size="invisible"
      />
      {status === "success" && (
        <p>
          {t('success', {fromNameValue: fromNameValue})}
        </p>
      )}
      {status === "error" && (
        <p>{t('error', {fromNameValue: fromNameValue})}</p>
      )}
      <div className="flex place-content-end">
        <TzButton
          iconName="ArrowRightIcon"
          isIconAfterLabel
          variant="accent"
          label={t('submit')}
        />
      </div>
    </form>
  );
}
