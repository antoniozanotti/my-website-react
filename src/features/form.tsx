import React from 'react';
import { ReactNode } from 'react';

export interface FormProps {
  
}

export const Form = ({
  
}: FormProps) => {
  return (
    <form id="form_lets_work_together">
      <label>
        Name
        <input
          type="text"
          name="inp_name"
          id="form_lets_work_together__inp_name"
        />
      </label>
      <label>
        Email
        <input type="text" name="inp_email" />
      </label>
      <label>
        Message
        <textarea name="tex_message"></textarea>
      </label>
      <button type="submit">Submit</button>
    </form>
  );
};