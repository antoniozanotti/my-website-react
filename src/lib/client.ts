import { ApolloClient, HttpLink, InMemoryCache } from "@apollo/client";
import { registerApolloClient } from "@apollo/experimental-nextjs-app-support/rsc";

const TOKEN = process.env.CONTENTFUL_ACCESS_TOKEN;
const SPACE = process.env.CONTENTFUL_SPACE_ID;
const ENVIRONMENT = process.env.CONTENTFUL_ENVIRONMENT_ID;
const URL = `https://graphql.contentful.com/content/v1/spaces/${SPACE}/environments/${ENVIRONMENT}`;

export const { getClient } = registerApolloClient(() => {
  return new ApolloClient({
    cache: new InMemoryCache(),
    link: new HttpLink({
      uri: URL,
      headers: {
        Authorization: `Bearer ${TOKEN}`,
      },
    }),
  });
});
