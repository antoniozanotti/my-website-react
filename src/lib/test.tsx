import React, { ReactElement } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";
import { Theme } from "@/domain/theme/theme/theme";
import { Providers } from "../app/providers";
import messages from "@/messages/en.json";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return (
    <Providers locale="en" messages={messages}>
      <Theme>
        <div data-testid="container">{children}</div>
      </Theme>
    </Providers>
  );
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
  };
});
