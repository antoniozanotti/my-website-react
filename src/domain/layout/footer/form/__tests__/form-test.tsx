import { screen, render } from "@/lib/test";
import { Form } from "../form";

describe("Form", () => {
  let container: HTMLElement;
  beforeEach(() => {
    container = render(<Form />).container;
  });

  it("should render form", async () => {
    const nameField = screen.getByRole("textbox", { name: "Name" });
    const emailField = screen.getByRole("textbox", { name: "Email" });
    const messageField = screen.getByRole("textbox", { name: "Message" });
    const button = screen.getByRole("button");

    expect(container).toMatchSnapshot();
  });
});
