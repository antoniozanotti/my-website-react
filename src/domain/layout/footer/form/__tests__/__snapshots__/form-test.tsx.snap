// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render form 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <form
          class="flex flex-col space-y-4"
          id="form_lets_work_together"
        >
          <label
            class="text-primary-2 dark:text-dark-primary-2 flex flex-col space-y-1"
          >
            <span
              class="px-[10px] sm:px-[14px] lg:px-[20px]"
            >
              Name
            </span>
            <input
              class="rounded focus:outline focus:outline-offset-2 focus:outline-2 h-[28px] sm:h-[38px] lg:h-[48px] px-[10px] sm:px-[14px] lg:px-[20px]  bg-secondary dark:bg-dark-secondary hover:bg-secondary-1 active:bg-secondary-2 focus:outline-secondary dark:hover:bg-dark-secondary-1 dark:active:bg-dark-secondary-2 dark:focus:outline-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary"
              name="fromName"
              required=""
              type="text"
            />
          </label>
          <label
            class="text-primary-2 dark:text-dark-primary-2 flex flex-col space-y-1"
          >
            <span
              class="px-[10px] sm:px-[14px] lg:px-[20px]"
            >
              Email
            </span>
            <input
              class="rounded focus:outline focus:outline-offset-2 focus:outline-2 h-[28px] sm:h-[38px] lg:h-[48px] px-[10px] sm:px-[14px] lg:px-[20px]  bg-secondary dark:bg-dark-secondary hover:bg-secondary-1 active:bg-secondary-2 focus:outline-secondary dark:hover:bg-dark-secondary-1 dark:active:bg-dark-secondary-2 dark:focus:outline-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary"
              name="fromEmail"
              required=""
              type="email"
            />
          </label>
          <label
            class="text-primary-2 dark:text-dark-primary-2 flex flex-col space-y-1"
          >
            <span
              class="px-[10px] sm:px-[14px] lg:px-[20px]"
            >
              Message
            </span>
            <textarea
              class="rounded focus:outline focus:outline-offset-2 focus:outline-2 h-[177px] px-[10px] sm:px-[14px] lg:px-[20px] py-[7px] sm:py-[10px] lg:py-[12px]  bg-secondary dark:bg-dark-secondary hover:bg-secondary-1 active:bg-secondary-2 focus:outline-secondary dark:hover:bg-dark-secondary-1 dark:active:bg-dark-secondary-2 dark:focus:outline-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary"
              name="message"
              required=""
              rows="7"
            />
          </label>
          <div />
          <div
            class="flex place-content-end"
          >
            <button
              class="rounded-full font-medium flex gap-x-2 justify-center place-items-center focus:outline focus:outline-offset-2 focus:outline-2  bg-accent dark:bg-dark-accent hover:bg-accent-1 active:bg-accent-2 focus:outline-accent dark:hover:bg-dark-accent-1 dark:active:bg-dark-accent-2 dark:focus:outline-dark-accent text-inverse-accent dark:text-inverse-dark-accent h-[38px] lg:h-[48px] min-w-[38px] lg:min-w-[48px] px-[27px] lg:px-[36px] "
            >
              <svg
                aria-hidden="true"
                class="w-[16px] h-[16px] lg:w-[20px] lg:h-[20px] text-inverse-accent dark:text-inverse-dark-accent order-last"
                data-slot="icon"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  ArrowRightIcon
                </title>
                <path
                  d="M13.5 4.5 21 12m0 0-7.5 7.5M21 12H3"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
              <span>
                Submit
              </span>
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
